### DISCUSSION TITLE
Efficient Mixture of Experts implementation?

### DISCUSSION THREAD

**Discussion by SamKG** [google/flax#4035] [General] [answered]
Hello,

Is there an efficient implementation of Mixture of Experts models (e.g. Mixtral) anywhere that uses flax/jax? 
I can't seem to find one, though I'd be happy to try writing one myself if there's interest.

As an aside: is block-sparse matmul efficient in Jax/Flax?

**Comment by cgarciae** [comment, marked as answer]
Checkout the [moe](https://github.com/google/flaxformer/tree/main/flaxformer/architectures/moe) implementation from Flaxformers.
